be/src/pipeline/pipeline_fragment_context.cpp

```diff
@@ -1167,7 +1167,8 @@ Status PipelineFragmentContext::_create_operator(ObjectPool* pool, const TPlanNo
             op.reset(new StreamingAggOperatorX(pool, next_operator_id(), tnode, descs));
             RETURN_IF_ERROR(cur_pipe->add_operator(op));
         } else {
-            if (_runtime_state->enable_agg_spill() && !tnode.agg_node.grouping_exprs.empty()) {
+            if ((_runtime_state->enable_force_spill() || _runtime_state->enable_agg_spill()) &&
```

Comment:
DO not modify like this.
in enable_agg_spill() method add a check enable_force_spill